app/src/ingestion/pdf_postprocess.py

```diff
@@ -223,6 +223,59 @@ def _should_merge_text_split_across_pages(text: EnrichedText, next_text: Enriche
     return next_text.text.strip()[0].islower()
 
 
+def _find_heading_index(heading_list: list[str], heading_text: str) -> int:
+    try:
+        return heading_list.index(heading_text)
+    except ValueError:
+        return -1
+
+
+def _group_headings_text(markdown_texts: list[EnrichedText]) -> list[EnrichedText]:
+    grouped_texts_by_headings: list[EnrichedText] = []
+    first_paragraph = markdown_texts[0]
+    combined_text = ""
+    current_heading = first_paragraph.headings
+    page_num = first_paragraph.page_number
+    elem_type = first_paragraph.type
+    headings_processed: list[str] = []
+    for ind, markdown_text in enumerate(markdown_texts):
```

Comment:
Since this is several lines of code, add code comments to describe what is intended to be done by the code. Alternatively, extract small code blocks into well-named functions.